apps/site/assets/ts/schedule/schedule-loader.tsx

```diff
@@ -56,9 +52,33 @@ const renderSchedulePage = (): void => {
   }
 };
 
+const renderDirection = (schedulePageData: SchedulePageData): void => {
+  const {
+    direction_id: directionId,
+    route_patterns: routePatternsByDirection,
+    shape_map: shapesById,
+    route
+  } = schedulePageData;
+  ReactDOM.render(
+    <ScheduleDirection
+      directionId={directionId}
+      route={route}
+      routePatternsByDirection={routePatternsByDirection}
+      shapesById={shapesById}
+    />,
+    document.getElementById("react-schedule-direction-root")
+  );
+};
+
 const render = (): void => {
   renderMap();
-  renderSchedulePage();
+  const schedulePageDataEl = document.getElementById("js-schedule-page-data");
+  if (!schedulePageDataEl) return;
+  const schedulePageData = JSON.parse(
```

Comment:
pretty much. When using React (and possibly other similar frameworks) there are two approaches: 1) fetch the data after the component is loaded, 2) pre-populate the data on the page and only fetch additional data if/when that is necessary.

It may look a little odd to have some hidden element in the page that contains a JSON blob... but yeah, it's fairly typical :)

Reply:
we are doing option 2 because it has a better user experience